design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,474 @@
+
+- [Delivery Plan - Conjur Azure Authenticator](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Delivery+Plan+-+Conjur+Azure+Authenticator)
+
+## Issue description
+
+Many of our customers are using Azure as their main cloud provider. Therefore we want to enable Azure resources to authenticate according to Azure 
+properties and get back a Conjur access token in order to retrieve secrets from our providers. At current, customers can authenticate their Azure 
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the feature doc, before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+  
+  - !variable
+    id: provider-uri
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+and load a value to the `provider-uri` variable (e.g `https://sts.windows.net/TENANT_ID`). 
+
+### Azure Application Identity
+
+To authenticate with Conjur, Azure-specific fields will need to be provided in the host annotations of the Conjur host identity for the Azure resource- subscription-id, resource-group, and user/system-assigned-identity. Specifically, If the Azure resource is assigned a system-assigned-identity, 
+we will also need to validate the `oid` field of the Azure access token. These fields will be used to confirm the identity of the Azure resource that is
+sending the authentication request. This will be done by comparing the Azure access token information with the host annotations that are defined in the policy below.
+
+A Conjur host will be defined as follows with their annotations holding Azure-specific identification properties.
+
+```yaml
+- !policy 
+  id: azure-apps 
+  body: 
+  - !host 
+    id: test-app 
+    annotations: 
+      authn-azure/subscription-id: test-subscription 
+      authn-azure/resource-group: test-group 
+      authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+      authn-azure/system-assigned-identity: 853b9a84-5bfa-4b22-a3f3-0b9a43d9ad8a
+
+- !grant
+  role: !group conjur/authn-azure/prod/apps
+  member: !host azure-apps/test-app
+```
+
+Note that such a host can have only one of the annotations 
+`authn-azure/user-assigned-identity` & `authn-azure/system-assigned-identity`
+
+#### Multiple-assigned identity for host - NOT FOR NOW
```

Comment:
Is this planned for any future phase?

Reply:
Not really. Do you think it's better to remove this altogther? I believe that having it here let's the reader understand that this use-case was considered and was decided not to be implemented atm.